RxCocoa/iOS/Proxies/RxScrollViewDelegateProxy.swift

```diff
@@ -14,24 +14,18 @@ import RxSwift
 import UIKit
 
 /// For more information take a look at `DelegateProxyType`.
-public class RxScrollViewDelegateProxy
-    : DelegateProxy
-    , UIScrollViewDelegate
-    , DelegateProxyType {
+open class RxScrollViewDelegateProxy<P: UIScrollView>
+    : DelegateProxy<P, UIScrollViewDelegate>
```

Comment:
I think the `Delegate` should fix in a type declared `DelegateProxyType`.
`Delegate` affect `DelegateProxy.installForwardDelegate` implementation, and of course we can call `DelegateProxy.installForwardDelegate` from upper casted parent object.
```swift
// e.g.
let tableView = UITableView()
RxScrollViewDelegateProxy.installForwardDelegate(myScrollViewDelegate, retainDelegate: false, onProxyForObject: tableView)
tableView.rx.delegate.forwardToDelegate() // -> It is myScrollViewDelegate.
```

We know same pattern in UIKit, it seems strange.
```swift
let tableView = UITableView()
(tableView as UIScrollView).delegate = myScrollViewDelegate
tableView.delegate // -> myScrollViewDelegate, but it behave as UITableViewDelegate
```

Reply:
Hi @tarunon ,

I'm sorry but I don't think I understand what did you want to say here 😅 ? Is there some problem?

Reply:
I think it is no problem here. 
I mean it is difficult making `RxTableViewDelegateProxy<UITableView, UITableViewDelegate>`.
I'm sorry confused you.